packages/react-native/sdks/hermes-engine/hermes-utils.rb

```diff
@@ -298,18 +298,14 @@ def resolve_url_redirects(url)
 
 # [macOS react-native-macos does not publish macos specific hermes artifacts
 # so we attempt to find the latest patch version of the iOS artifacts and use that
-def findLatestVersionWithArtifact(version)
-    if version == "1000.0.0"
+def findLatestVersionWithArtifact(package)
+    if package['version'] == "1000.0.0"
         # The main branch builds from source, so skip the artifact check
         return nil
     end
 
-    # See https://central.sonatype.org/search/rest-api-guide/ for details on query params
-    versionWithoutPatch = "#{version.match(/^(\d+\.\d+)/)}"
-    res, = Open3.capture3("curl -s https://search.maven.org/solrsearch/select?q=g:com.facebook.react+AND+a:react-native-artifacts+AND+v:#{versionWithoutPatch}.*&core=gav&rows=1&wt=json")
-    wt = JSON.parse(res)
-    response = wt['response']
-    return response['docs'][0]['v'] unless response['numFound'] == 0
+    # This assumes that whatever peer dependency we specify has an artifact
+    return package['peerDependencies']['react-native']
```

Comment:
Maybe we can keep the previous logic as a fallback if we don't have a peer dependency specified? Theoretically we should always have one after this set of changes lands, but an idea.